python/dolfinx/fem/function.py

```diff
@@ -412,6 +412,54 @@ def eval(self, x: npt.ArrayLike, cells: npt.ArrayLike, u=None) -> np.ndarray:
             u = np.reshape(u, (-1,))
         return u
 
+    def interpolate_nonmatching_meshes(
+        self,
+        u: Function,
+        cells: typing.Optional[np.ndarray] = None,
+        nmm_interpolation_data: typing.Optional[PointOwnershipData] = None,
+        padding: float = 0.0,
+    ) -> None:
+        """Interpolate a function defined on one mesh to a function defined on another mesh
+
+        Args:
+            u: The Function to interpolate.
+            cells: The cells to interpolate over. If `None` then all
+                cells are interpolated over.
+            nmm_interpolation_data: Data needed to interpolate functions defined on other meshes
+            padding: Absolute padding of bounding boxes for computing
```

Comment:
Force user to create `nmm_interpolation_data`, then this isn't needed. 

We should avoid redundant arguments - if a user passes `nmm_interpolation_data` and `padding`, then `padding` has no effect. Need to keep the logic simple.